olp-cpp-sdk-authentication/src/SignInResultImpl.cpp

```diff
@@ -91,5 +103,63 @@ const std::string& SignInResultImpl::GetScope() const { return scope_; }
 
 bool SignInResultImpl::IsValid() const { return is_valid_; }
 
+std::string SignInResultImpl::ParseJwtToken(const std::string& token) {
+  std::string result;
+  std::string::size_type first_dot = token.find_first_of('.');
+  if (std::string::npos == first_dot) {
+    OLP_SDK_LOG_ERROR_F(kLogTag, "Cannot parse ClientId. Wrong token format.");
+    return result;
+  }
+
+  std::string jws_header_encoded = token.substr(0, first_dot);
+
+  std::vector<std::uint8_t> decoded_bytes;
+  if (!utils::Base64Decode(jws_header_encoded, decoded_bytes)) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
+                        "Cannot parse ClientId. Non-decodable token format");
+    return result;
+  }
+
+  std::string jws_header_decoded;
+  std::copy(decoded_bytes.begin(), decoded_bytes.end(),
+            std::back_inserter(jws_header_decoded));
+
+  rapidjson::Document doc;
+  doc.Parse(jws_header_decoded.c_str());
+
+  if (doc.HasParseError() || !doc.HasMember("aid")) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
+                        "Cannot parse ClientId. Defective token format");
+    return result;
+  }
+
+  result = doc["aid"].GetString();
```

Comment:
```suggestion
const std::string result = doc["aid"].GetString();
```
See my previous comment about result variable